tutorial/beginner/t11_debugging.ipynb

```diff
@@ -0,0 +1,854 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a965111b",
+   "metadata": {},
+   "source": [
+    "# Tutorial 11: Debugging"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d7b10da",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "In this tutorial we are going to cover:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "609140d0",
+   "metadata": {},
+   "source": [
+    "* [Pipeline Debugging](#t11PipelineDebugging)\n",
+    "    * [Debug the Single NumpyOp](#t11SingleNumpyop)\n",
+    "    * [Debug and Verify the Results of Pipeline](#t11VerifyPipeline)\n",
+    "* [Network Debugging](#t11NetworkDebugging)\n",
+    "    * [Debug the Single TensorOp](#t11SingleTensorOp)\n",
+    "    * [Debug and Verify the Results of Network](#t11VerifyNetwork)\n",
+    "* [Trace Debugging](#t11TraceDebugging)\n",
+    "    * [Conditional Debugging in Training](#t11Conditional)\n",
+    "    * [Debug Pipeline and Network From the Trace](#t11DebugPiplineNetwork)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f5f618f0",
+   "metadata": {},
+   "source": [
+    "## Pipeline Debugging"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a322ba51",
+   "metadata": {},
+   "source": [
+    "In [Tutorial4](./t04_pipeline.ipynb) we demonstrated what Pipeline is and how it is created to handle different preprocessing tasks using `NumpyOp`. Since the pipeline consists of series of NumpyOps, it's a vital to know how to debug the `NumpyOp`. \n",
```

Comment:
```suggestion
    "In [Tutorial 4](./t04_pipeline.ipynb) we demonstrated what the Pipeline is and how it is created to handle different preprocessing tasks using `NumpyOp`s. Since the pipeline consists of a series of NumpyOps, it's a vital to know how to debug those `NumpyOp`s. \n",
```